### DISCUSSION TITLE
How can I specify exact GELF endpoint?

### DISCUSSION THREAD

**Discussion by prcdpr** [dustinchilson/NLog.Targets.GraylogHttp#115] [Q&A] [answered]
Let's say I would like to specify exact GELF endpoint to be:
```
https://graylog.example.org:[port]/gelf
 ```

Is it currently possible with `graylogServer` and `graylogPort` XML attributes?
There might be a scenario when Graylog server is behind reverse proxy and it isn't `/gelf` on the root level of the path.

**Comment by dustinchilson** [comment, marked as answer]
`/gelf` is always appended to the url,

https://github.com/dustinchilson/NLog.Targets.GraylogHttp/blob/master/src/NLog.Targets.GraylogHttp/GraylogHttpTarget.cs#L64

So if you want to move it to a subfolder behind a proxy you can provide the port (Assuming its a non-standard port for Http[s]) as part of the `graylogServer` variable and not provide the `graylogPort` variable.

The Port is optional, https://github.com/dustinchilson/NLog.Targets.GraylogHttp/blob/master/src/NLog.Targets.GraylogHttp/GraylogHttpTarget.cs#L66

**Comment by prcdpr** [comment]
Thanks!
It solves my problem because I have subfolder in URL but there is still `/gelf` ending.
Someone else may have different URL so maybe it should not be hardcoded?

**Comment by dustinchilson** [comment]
It's part of the Graylog Http Input spec, 

https://docs.graylog.org/en/3.1/pages/gelf.html#sending-gelf-messages-via-http-using-curl

But it might be true that in some proxy situations that it might not be great. Might be helpful to have a setting to disable that.
